lib/src/inline_parser.dart

```diff
@@ -92,6 +93,7 @@ class InlineParser {
     if (document.withDefaultInlineSyntaxes) {
       // Custom link resolvers go after the generic text syntax.
       syntaxes.addAll([
+        DecodeHtmlSyntax(),
```

Comment:
The addition to the changelog says

> * Add a new syntax `DecodeHtmlSyntax` to decode HTML entity and numeric
  character references.

However, this changes the default as well, it's definitely an incompatible change. Was that intended?

And, how to opt out of this change, if necessary?

Reply:
If you need to run the parser without `DecodeHtmlSyntax`, you have to set `withDefaultInlineSyntaxes` as `false` and add all the inline syntaxes you want to have through `inlineSyntaxes` option,  unfortunately there is not a better way at this moment. Maybe we could add options to make it possible to disable each individual syntax in the future.

Reply:
The reason that I'm inquiring is that this change makes the following test fail:

https://github.com/flutter/packages/blob/8496fe847adfdf64d0779dc683d2bbb93e960ba8/packages/flutter_markdown/test/html_test.dart#L56

and it is hard for me to judge whether that test's expectation is important for their purposes or not.

Reply:
Also, flutter_markdown may need to bend towards the spec. We have users writing markdown that shows up on GitHub readme rendering. We should be mindful that we can't have bespoke one-off interpretations of the spec – it'll just annoy our users.

Reply:
@stuartmorgan - who can speak to the package:flutter_markdown behavior / impl? package:markdown's impl is changing so that the flutter_markdown test `doesn't convert existing HTML entities when parsing` will no longer be true.

(we can also track this as in issue in flutter/flutter if that's easier for the conversation)

Reply:
I'm not very familiar with the details of `flutter_markdown` (/cc @domesticmouse who does most of the stewardship for it), but its documented behavior is to follow the relevant specs, so if the test is going against the spec I would think we should just fix the test.

Reply:
I'm owner in as much as I make sure changes have test coverage. I don't have any feelings about compliance against the various flavours of Markdown standards (CommonMark, GitHub flavoured, et al). 

I'm happy for tests to be fixed in the direction of standards compliance...

Reply:
I filed this to track removing the test: https://github.com/flutter/flutter/issues/114193 (and https://github.com/dart-lang/markdown/issues/477 for some additional pre-submit checking for package:markdown).

Reply:
https://github.com/flutter/packages/pull/2757